src/Helpers/SAMLUserGroupMapper.php

```diff
@@ -0,0 +1,65 @@
+<?php
+
+namespace SilverStripe\SAML\Helpers;
+
+use SilverStripe\Core\Config\Configurable;
+use SilverStripe\Core\Injector\Injectable;
+use SilverStripe\ORM\DataObject;
+use SilverStripe\SAML\Services\SAMLConfiguration;
+use SilverStripe\Security\Group;
+use SilverStripe\Security\Member;
+
+class SAMLUserGroupMapper
+{
+    use Injectable;
+    use Configurable;
+
+    /**
+     * @var string
+     */
+    private static $group_claims_field;
+
+    /**
+     * @var array
+     */
+    private static $dependencies = [
+        'SAMLConfService' => '%$' . SAMLConfiguration::class,
+    ];
+
+    /**
+     * Check if group claims field is set and assigns member to group
+     *
+     * @param [] $attributes
+     * @param Member $member
+     * @return Member
+     */
+    public function map($attributes, $member): Member
+    {
+        $groups = $this->config()->get('group_claims_field');
+
+        if (!isset($attributes[$groups])) {
+            return $member;
+        }
+
+        // Get groups from saml response
+        $groupTitles = $attributes[$groups];
+
+        foreach ($groupTitles as $groupTitle) {
+            // Get Group object by Title
+            $group = DataObject::get_one(Group::class, [
+                '"Group"."Title"' => $groupTitle
+            ]);
+
+            // Create group if it doesn't exist yet
+            if (!$group) {
+                $group = new Group();
+                $group->Title = $groupTitle;
+                $group->write();
+            }
+
+            $member->Groups()->add($group);
```

Comment:
I think we also want to remove users from groups they are no longer part of - this probably requires marking groups as 'synced' in some way so we know which ones to remove them from. Its quite likely that a site would have some groups synced with saml but others that are not (we've got at least one project with this requirement) - we wouldn't want to remove users from these 'manual' groups.